### PR TITLE
KotOR savegames WIP

### PR DESCRIPTION
**Pull request by seedhartha**
Made load/save game menu scrollable using up/down arrows; implemented saved game loading (time played, current module, PC gender and position)

### CODE REVIEW

**Comment by DrMcCoy** [comment]
Frankly, I'm not sure loading games is such a good idea yet. Can we already evaluate all the saved state? And it kinda pretends we already have proper gameplay.

Loading a savegame is something I'd go for far later, once we already have all the gameplay in place. Combat, conversations, script triggers, etc.

**Comment by seedhartha** [comment]
Sure. Wasn't going for a full blown implementation anyway :) Only had current module, PC gender, class, head and position loading planned.

**Comment by seedhartha** [comment]
Implemented current module and PC position loading. Merge would be appreciated despite low task priority :)

**Comment by seedhartha** [comment]
Implemented current module and PC position loading.

**Comment by DrMcCoy** [comment]
Trying it out with a few of my savegames...
1) The numbering is different in the original
2) The loading of the PC head and skincolor is wrong. Your PR reads different values than the original game

**Comment by DrMcCoy** [comment]
No, wait, the heads and skincolors are even random. Oh, you're not yet loading that, I guess? :P

**Comment by seedhartha** [comment]
Yep, thought that would be enough for now.

File: src/engines/aurora/gui.h [hunk 1]
```diff
@@ -63,6 +63,12 @@ class GUI : boost::noncopyable {
 	/** Process the current event queue. */
 	uint32 processEventQueue();
 
+	/** Add a widget.
+	 *
+	 *  @param show Show it immediately?
+	 */
+	void addWidget(Widget *widget, bool show = false);
```
Comment (DrMcCoy) [inline src/engines/aurora/gui.h]: I don't have time today to look deeper into it, but is this really necessary? I don't like having methods that change the GUI itself public.

**Comment by seedhartha** [comment]
Made addWidget private again. Also implemented PC gender loading.

**Comment by DrMcCoy** [comment]
Merged, thanks! :)

**Pull request merged**